arch/SH/SHDisassembler.c

```diff
@@ -2163,7 +2163,7 @@ static bool sh_disassemble(const uint8_t *code, MCInst *MI, uint64_t address,
 		idx = ((insn >> 8) & 0xf0) | (insn & 0x000f);
 	}
 
-	if (decode[idx]) {
+	if (idx < sizeof(decode)/sizeof(decode[0]) &&  decode[idx]) {
```

Comment:
Please use the macro `ARR_SIZE` from `utils.h` for this:

```suggestion
	if (idx < ARR_SIZE(decode) && decode[idx]) {
```